NuRadioReco/modules/io/coreas/coreas.py

```diff
@@ -256,3 +415,517 @@ def make_sim_shower(corsika, observer=None, detector=None, station_id=None):
+
+        Parameters
+        ----------
+        position_on_ground : np.array (3)
+            position of the antenna on ground
+
+        core : np.array (3)
+            position of the core on ground
+
+        Returns
+        -------
+        efield_interp : float
+            interpolated efield value or efield of clostest observer
+
+        trace_start_time : float
+            start time of the trace
+        """
+        logger.debug(f"get interpolated efield for antenna position {position_on_ground} on ground and core position {core}")
+        antenna_position = copy.copy(position_on_ground)
+        #core and antenna need to be in the same z plane
+        antenna_position[2] = core[2]
+        # transform antenna position into shower plane with respect to core position, core position is set to 0,0 in shower plane
+        antenna_pos_vBvvB = self.cs.transform_to_vxB_vxvxB(antenna_position, core=core)
+        logger.debug(f"antenna position in shower plane {antenna_pos_vBvvB}")
+
+        # calculate distance between core position at(0,0) and antenna positions in shower plane
+        dcore_vBvvB = np.linalg.norm(antenna_pos_vBvvB[:-1])
+        # interpolate electric field at antenna position in shower plane which are inside star pattern
+        if dcore_vBvvB > self.star_radius:
+            efield_interp = self.empty_efield
+            logger.debug(f'antenna position with distance {dcore_vBvvB:.2f} to core is outside of star pattern with radius {self.star_radius:.2f} on ground {self.geo_star_radius:.2f}, set efield and fluence to zero')
+        else:
+            if self.efield_interpolator == -1:
+                efield = self.get_closest_observer_efield(antenna_pos_vBvvB)
+                efield_interp = efield
+                trace_start_time = None
+
+            else:
+                efield_interp, trace_start_time, abs_spectrum, phasespectrum = self.efield_interpolator(antenna_pos_vBvvB[0], antenna_pos_vBvvB[1],
+                                                lowfreq=self.interp_lowfreq/units.MHz,
+                                                highfreq=self.interp_highfreq/units.MHz,
+                                                filter_up_to_cutoff=False,
+                                                account_for_timing=True,
+                                                pulse_centered=True,
+                                                const_time_offset=20.0e-9,
+                                                full_output=True)
+
+        #check if interpolation is within expected range
+        if np.max(np.abs(efield_interp)) > self.max_coreas_efield:
+            logger.warning(f'interpolated efield {np.max(np.abs(efield_interp)):.2f} is larger than the maximum coreas efield {self.max_coreas_efield:.2f}')
+        return efield_interp, trace_start_time
```

Comment:
`trace_start_time` is not defined if the `dcore_vBvvB > self.star_radius` condition is met

Reply:
Thanks!